ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java

```diff
@@ -99,4 +108,41 @@ private static Executable createExecutable(
     builder.put(StreamsConfig.APPLICATION_ID_CONFIG, KSQL_REST_SERVER_DEFAULT_APP_ID);
     return builder.build();
   }
+
+  @VisibleForTesting
+  static void enforceStreamStateDirAvailability(final File streamsStateDir) {
+    if (!streamsStateDir.exists()) {
+      final boolean mkDirSuccess = streamsStateDir.mkdir();
+      if (!mkDirSuccess) {
+        throw new KsqlServerException("Could not create the kafka streams state directory: "
+            + streamsStateDir.getPath()
+            + "\n Make sure the directory exists and is writable for KSQL server "
+            + "\n or its parend directory is writbale by KSQL server"
+            + "\n or change it to a writable directory by setting '"
+            + KsqlConfig.KSQL_STREAMS_PREFIX + StreamsConfig.STATE_DIR_CONFIG
+            + "' config in the properties file."
+        );
+      }
+    }
+    if (!streamsStateDir.isDirectory()) {
+      throw new KsqlServerException(streamsStateDir.getPath()
+          + " is not a directory."
+          + "\n Make sure the directory exists and is writable for KSQL server "
+          + "\n or its parend directory is writbale by KSQL server"
+          + "\n or change it to a writable directory by setting '"
+          + KsqlConfig.KSQL_STREAMS_PREFIX + StreamsConfig.STATE_DIR_CONFIG
+          + "' config in the properties file."
+      );
+    }
+    if (!streamsStateDir.canWrite()) {
```

Comment:
On unix systems this also needs to check for execute permissions. I'm not sure what this translates to for Windows. A more robust check would be to just try creating a file and making sure it shows up in a listing.

Reply:
Checking both writable and executable now.